substrate/frame/beefy/src/equivocation.rs

```diff
@@ -88,7 +90,7 @@ where
 	type TimeSlot = TimeSlot<N>;
 
 	fn offenders(&self) -> Vec<Offender> {
-		vec![self.offender.clone()]
+		self.offenders.clone()
```

Comment:
It's returning a `Vec` since required by the underlying `Offence` trait:
https://github.com/lederstrumpf/polkadot-sdk/blob/e7718fc8e75ed7133b3c06061374f32beefa7b8f/substrate/primitives/staking/src/offence.rs#L81

Changing this has a few downstream effects, such as for babe, grandpa
https://github.com/lederstrumpf/polkadot-sdk/blob/e7718fc8e75ed7133b3c06061374f32beefa7b8f/substrate/frame/babe/src/equivocation.rs#L75-L77
https://github.com/lederstrumpf/polkadot-sdk/blob/e7718fc8e75ed7133b3c06061374f32beefa7b8f/substrate/frame/grandpa/src/equivocation.rs#L85-L87
and for the Offenders pallet
https://github.com/lederstrumpf/polkadot-sdk/blob/e7718fc8e75ed7133b3c06061374f32beefa7b8f/substrate/frame/offences/src/lib.rs#L158-L163

Had an initial stab at it (https://github.com/Lederstrumpf/polkadot-sdk/commit/2d25453cfed988112b8115087436680f49f26927), but ended up with quite a few places to change that would require review from a few other code-owners, and haven't gone down the can of worms that this breaks e.g. for `OffenceDetails`. So I'd suggest doing that in a followup-PR, if at all.

---

Otherwise, would you be happier with
```suggestion
		self.offenders.as_slice().to_vec()
```
?
Performance should be the same.

Reply:
Oh sorry, I didn't realize this was part of a trait defined in another pallet. We can keep it as it is then.